ale_linters/c/clang.vim

```diff
@@ -8,22 +8,33 @@ function! ale_linters#c#clang#GetExecutable(buffer) abort
     return ale#Var(a:buffer, 'c_clang_executable')
 endfunction
 
-function! ale_linters#c#clang#GetCommand(buffer) abort
-    let l:paths = ale#c#FindLocalHeaderPaths(a:buffer)
+function! ale_linters#c#clang#GetCommand(buffer, output) abort
+    let l:cflags = []
+    if !empty(a:output)
+        let l:cflags = join(ale#c#ParseMakefile(a:buffer, join(a:output, '\n')), ' ')
+    endif
+    if empty(l:cflags)
+        let l:cflags = ale#c#IncludeOptions(ale#c#FindLocalHeaderPaths(a:buffer))
+    else
+        let l:cflags .= ' '
```

Comment:
You can't use `.= ' '` with a List. The variable should be either a String or a List, but not both.